### PR TITLE
Code improvements to various classes to remove unused and unnecessary code

### PR DESCRIPTION
**Pull request by Matthew-Bonner**
Code improvements to various classes to remove unused and unnecessary code

### CODE REVIEW

**Comment by Matthew-Bonner** [comment]
Rather than squashing the commits, it makes sense in my view to keep the commits as because I'm working through the files one by one.

**Comment by kinglozzer** [comment]
Hey @Matthew-Bonner. It looks like these `private static` properties aren’t being used, but they’re actually picked up by the Config system and used in a few places. To take the first example from the test failures: [this property](https://github.com/silverstripe/silverstripe-cms/blob/3.1/code/controllers/CMSMain.php#L26) is accessed [here](https://github.com/silverstripe/silverstripe-cms/blob/3.1/code/controllers/CMSMain.php#L512). The `$allowed_actions` are used by `RequestHandler` to determine which URLs can be accessed on a Controller too.

The other bits of tidying up (removing unused variables from methods, deleting commented-out code etc) seem fine, though. It might be easier to close this, start afresh and open a new pull request, rather than undo those `private static` deletions. But I’ll leave that up to you :)

File: code/controllers/CMSMain.php [hunk 1]
```diff
@@ -1196,16 +1150,23 @@ public function BatchActionList() {
 	
 	public function buildbrokenlinks($request) {
 		// Protect against CSRF on destructive action
-		if(!SecurityToken::inst()->checkRequest($request)) return $this->httpError(400);
+		if(!SecurityToken::inst()->checkRequest($request)) {
+      $this->httpError(400);
+      return;
+    }
```
Comment (kinglozzer) [inline code/controllers/CMSMain.php]: We actually don’t even need the `return` statement here as `RequestHandler::httpError()` throws an `SS_HTTPResponse_Exception`, so it’s redundant. We could keep this on one line and just have:
```php
if(!SecurityToken::inst()->checkRequest($request)) $this->httpError(400);
```

File: code/controllers/CMSMain.php [hunk 2]
```diff
@@ -1196,16 +1150,23 @@ public function BatchActionList() {
...
 		
 		increase_time_limit_to();
 		increase_memory_limit_to();
-		
+
+    $newPageSet = array();
```
Comment (kinglozzer) [inline code/controllers/CMSMain.php]: Tabs for indentation instead of spaces please :)

File: code/controllers/CMSMain.php [hunk 3]
```diff
@@ -1348,7 +1315,10 @@ public function duplicate($request) {
 
 	public function duplicatewithchildren($request) {
 		// Protect against CSRF on destructive action
-		if(!SecurityToken::inst()->checkRequest($request)) return $this->httpError(400);
+		if(!SecurityToken::inst()->checkRequest($request)) {
+      $this->httpError(400);
+      return null;
+    }
```
Comment (kinglozzer) [inline code/controllers/CMSMain.php]: Same as above for these `httpErrors()`

**Comment by Matthew-Bonner** [comment]
No problem, I'll change the settings on the IDE and check what is being used then I'll submit a new pull request. Thanks for spotting that some are used, I'll try and be a little more careful in working out what is used and what isn't.

File: code/controllers/AssetAdmin.php [hunk 4]
```diff
@@ -25,28 +16,6 @@ class AssetAdmin extends LeftAndMain implements PermissionProvider{
 	private static $page_length = 15;
 	
 	/**
-	 * @config
-	 * @see Upload->allowedMaxFileSize
-	 * @var int
-	 */
-	private static $allowed_max_file_size;
```
Comment (chillu) [inline code/controllers/AssetAdmin.php]: That removes the config hint which we might start parsing into something useful soon

File: code/controllers/CMSMain.php [hunk 5]
```diff
@@ -1196,16 +1150,23 @@ public function BatchActionList() {
...
+
+    $newPageSet = array();
+
 		if($this->urlParams['ID']) {
 			$newPageSet[] = DataObject::get_by_id("Page", $this->urlParams['ID']);
 		} else {
 			$pages = DataObject::get("Page");
-			foreach($pages as $page) $newPageSet[] = $page;
+			foreach($pages as $page) {
+        $newPageSet[] = $page;
```
Comment (chillu) [inline code/controllers/CMSMain.php]: Indentation as well

File: code/controllers/AssetAdmin.php [hunk 6]
```diff
@@ -7,15 +7,6 @@
...
  */
 class AssetAdmin extends LeftAndMain implements PermissionProvider{
-
-	private static $url_segment = 'assets';
-	
-	private static $url_rule = '/$Action/$ID';
-	
-	private static $menu_title = 'Files';
-
-	private static $tree_class = 'Folder';
```
Comment (chillu) [inline code/controllers/AssetAdmin.php]: Are you sure this isn't used?

File: code/controllers/AssetAdmin.php [hunk 7]
```diff
@@ -25,28 +16,6 @@ class AssetAdmin extends LeftAndMain implements PermissionProvider{
...
-		'DeleteItemsForm',
-		'SearchForm',
-		'getsubtree',
-		'movemarked',
-		'removefile',
-		'savefile',
-		'deleteUnusedThumbnails' => 'ADMIN',
-		'doSync',
-		'filter',
-	);
```
Comment (chillu) [inline code/controllers/AssetAdmin.php]: So by removing the whitelist, we allow all public methods? And `deleteUnusedThumbnails` becomes available to everybody with access to the controller?

**Comment by dhensby** [comment]
I'm not sure this kind of work (large scale removal of variables) should go on 3.1

If it's just redundant code removal, rather than declared class variables, then fine, otherwise it's 4.0

**Pull request closed without merging**